providers/test/Test.fst

```diff
@@ -359,126 +374,347 @@ let test_dh () : St unit =
-  end
+let external_test_set =
+  ts_nil
+  `ts_snoc` (OpenSSL, f_none)
+  `ts_snoc` (BCrypt, f_none)
 
-#pop-options
+inline_for_extraction
+noextract
+let poly1305_test_set =
+  (Hacl, f_avx2) `ts_cons` (
+  (Hacl, f_avx) `ts_cons` (
+  (Hacl, f_none) `ts_cons` (
+  (Vale, f_none) `ts_cons` (
+  external_test_set))))
 
 inline_for_extraction
 noextract
-let test_all_on_platform_avx (p: platform) (avx: bool) : St unit =
-  test_all_on_config p avx false;
-  test_all_on_config p avx true
+let curve25519_test_set =
+  (Hacl, f_bmi2 `f_concat` f_adx) `ts_cons` (
+  (Hacl, f_none) `ts_cons`
+  external_test_set)
 
 inline_for_extraction
 noextract
-let test_all_on_platform (p: platform) : St unit =
-  test_all_on_platform_avx p false;
-  test_all_on_platform_avx p true
+let aead_gcm_test_set =
+  (Vale, f_aesni `f_concat` f_avx) `ts_cons` (
+  ts_nil)
+
+inline_for_extraction
+noextract
+let chacha20poly1305_test_set =
+  (Hacl, f_none) `ts_cons`
+  external_test_set
+
+inline_for_extraction
+noextract
+let hash_test_set =
+  (Vale, f_none) `ts_cons` (
+  (Vale, f_shaext) `ts_cons` (
+  (Hacl, f_none) `ts_cons` (
+  external_test_set)))
+
+inline_for_extraction
+noextract
+let mk_test_set_with_none_avx_avx2 (i: impl) : Tot test_set =
+  (i, f_none) `ts_cons` (
+  (i, f_avx) `ts_cons` (
+  (i, f_avx2) `ts_cons`
+  ts_nil))
+
+inline_for_extraction
+noextract
+let default_test_set =
+  mk_test_set_with_none_avx_avx2 Vale `ts_append`
+  mk_test_set_with_none_avx_avx2 Hacl `ts_append`
+  external_test_set
+
+(* Test bodies *)
+
+inline_for_extraction
+noextract
+let test_poly1305_body (print: C.String.t -> St unit) : St unit =
+    print !$"  >>>>>>>>> Poly1305\n";
+    test_poly1305 ()
+
+inline_for_extraction
+noextract
+let test_curve25519_body (print: C.String.t -> St unit) : St unit =
+    print !$"  >>>>>>>>> Curve25519\n";
+    test_curve25519 ()
+
+inline_for_extraction
+noextract
+let test_aead_gcm_body (print: C.String.t -> St unit) : St unit =
+    print !$"  >>>>>>>>> AEAD (old vectors)\n";
+    test_aead ();
+    print !$"  >>>>>>>>> AEAD (AES128_GCM vectors)\n";
+    test_aes128_gcm ();
+    print !$"  >>>>>>>>> Cipher\n";
+    test_cipher ()
+
+inline_for_extraction
+let test_chacha20poly1305_body (print: C.String.t -> St unit) : St unit =
+    print !$"  >>>>>>>>> AEAD (ChachaPoly vectors)\n";
+    test_chacha20poly1305 ()
+
+inline_for_extraction
+noextract
+let test_hash_body (print: C.String.t -> St unit) : St unit =
+    print !$"  >>>>>>>>> Hash (Test.Hash)\n";
+    Test.Hash.main ();
+    print !$"  >>>>>>>>> Hash (Test.NoHeap)\n";
+    test_hash hash_vectors_low;
+    print !$"  >>>>>>>>> Hmac (Test.NoHeap)\n";
+    test_hmac hmac_vectors_low;
+    print !$"  >>>>>>>>> HKDF (Test.NoHeap)\n";
+    test_hkdf hkdf_vectors_low
+
+inline_for_extraction
+noextract
+let test_default_body (print: C.String.t -> St unit) : St unit =
+    print !$"  >>>>>>>>> FINITE-FIELD DIFFIE-HELLMAN\n";
+    test_dh ();
```

Comment:
this test is disabled and doesn't do anything; can we remove it?

Reply:
Thanks for confirming. Will do.